libcontainer/cgroups/fs/apply_raw.go

```diff
@@ -142,11 +142,19 @@ func (m *Manager) Apply(pid int) (err error) {
 			}
 			return err
 		}
-		m.Paths[sys.Name()] = p
 
 		if err := sys.Apply(d); err != nil {
+			if os.IsPermission(err) && m.Cgroups.Path == "" {
+				// If we didn't set a cgroup path, then let's defer the error here
+				// until we know whether we have set limits or not.
+				// If we hadn't set limits, then it's ok that we couldn't join this cgroup, because
+				// it will have the same limits as its parent.
+				continue
+			}
 			return err
 		}
+		m.Paths[sys.Name()] = p
```

Comment:
IIRC we moved this before `Apply` to handle some race conditions, maybe @hqhq remembers this better than me.